### PR TITLE
Reserv controller

### PR DESCRIPTION
**Pull request by acolombo1**
# Back End

## This PR includes the following:

Reservation controller:
Index action (get reservations)
Create action (post reservations)
(Show action not needed)

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class ReservationsController < ApplicationController
+  def create
+    myhash = params.require(:reservation).permit(:city, :date, :item)
+    myitem = Item.where(name: myhash['item']).first
+    new_reserv = Reservation.new(city: myhash['city'], date: myhash['date'], item: myitem)
+    new_reserv.user = current_user
```
Comment (alexpuente0) [inline app/controllers/reservations_controller.rb:8]: Great approach with the hash to hold the params!

File: app/controllers/reservations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class ReservationsController < ApplicationController
+  def create
+    myhash = params.require(:reservation).permit(:city, :date, :item)
```
Comment (alexpuente0) [inline app/controllers/reservations_controller.rb]: Please add authentication so only logged users can create a reservation

**Review by alexpuente0 — COMMENTED** [review]
Great work, do you think we should authenticate users so only logged in users can book a reservation?

File: app/controllers/reservations_controller.rb (hunk 2 above)
Comment (acolombo1) [reply app/controllers/reservations_controller.rb]: done!

File: app/controllers/reservations_controller.rb (hunk 1 above)
Comment (acolombo1) [reply app/controllers/reservations_controller.rb:8]: The way I found to split it

**Review by alexpuente0 — APPROVED** [review]
i see that you followed my suggestion and also made some additional improvements.
Everything checks out so you can merge this code

**Pull request merged**